Python/ceval.c

```diff
@@ -2163,9 +2163,22 @@ _PyEval_UnpackIterableStackRef(PyThreadState *tstate, _PyStackRef v_stackref,
             return 1;
         }
         Py_DECREF(w);
-        _PyErr_Format(tstate, PyExc_ValueError,
-                      "too many values to unpack (expected %d)",
-                      argcnt);
+
+        /* If v is a sequence or mapping, we can show its length in the error */
+        ll = PySequence_Size(v);
+        if (ll < 0) {
+            ll = PyMapping_Size(v);
+        }
```

Comment:
Oops I should've suggested calling `PyObject_Size` instead.
```suggestion
        ll = PyObject_Size(v);
```